src/fw/drivers/nrf5/rtc.c

```diff
@@ -70,20 +114,100 @@ RtcTicks rtc_get_ticks(void) {
   return s_coarse_ticks + rtc_interval_ticks;
 }
 
-void rtc_alarm_init(void) {
+/***
+ * Logic associated with converting extended RTC ticks to wall clock time.
+ */
+
+//! This variable is a UNIX timestamp of what the current wall clock time was at tick s_time_tick_base.
+static time_t s_time_base = 0;
+//! This variable is the tick where the wall clock time was equal to s_time_base. If you subtract this variable
+//! from the current tick count, you'll get the number of ticks that have elapsed since s_time_base, which will
+//! allow you to calculate the current wall clock time. Note that this value may be negative on startup, see
+//! restore_rtc_time_state
+static int64_t s_time_tick_base = 0;
+
+static time_t ticks_to_time(RtcTicks ticks) {
+  return s_time_base + ((ticks - s_time_tick_base) / RTC_TICKS_HZ);
 }
 
-void rtc_alarm_set(RtcTicks num_ticks) {
+void rtc_set_time(time_t time) {
+#ifdef PBL_LOG_ENABLED
+  char buffer[TIME_STRING_BUFFER_SIZE];
+  PBL_LOG(LOG_LEVEL_INFO, "Setting time to %lu <%s>", time, time_t_to_string(buffer, time));
+#endif
+
+  s_time_base = time;
+  s_time_tick_base = rtc_get_ticks();
+
+  save_rtc_time_state(s_time_tick_base - s_coarse_ticks);
 }
 
-RtcTicks rtc_alarm_get_elapsed_ticks(void) {
-  return 0;
+time_t rtc_get_time(void) {
+  return ticks_to_time(rtc_get_ticks());
 }
 
-bool rtc_alarm_is_initialized(void) {
-  return 0;
+void rtc_get_time_ms(time_t* out_seconds, uint16_t* out_ms) {
+  RtcTicks ticks = rtc_get_ticks();
+
+  RtcTicks ticks_since_time_base = (ticks - s_time_tick_base);
+  *out_seconds = s_time_base + (ticks_since_time_base / RTC_TICKS_HZ);
+
+  RtcTicks ticks_this_second = ticks_since_time_base % RTC_TICKS_HZ;
+  *out_ms = (ticks_this_second * 1000) / RTC_TICKS_HZ;
+}
+
+/***
+ * Logic associated with saving the RTC-tick-to-wallclock conversion factor
+ * to retained-RAM.
+ */
+
+static void restore_rtc_time_state(void) {
```

Comment:
`prv_` prefix

Reply:
In many instances!  This was mostly taken from stm32 rtc.c, which I think was written before this sort of style was enforced on PebbleOS.  Corrected all static functions to `prv_`.